### PR TITLE
ARC-1369 log serializer

### PR DESCRIPTION
**Pull request by joshkay10**
**Whats in the PR**

- The bulk of the files are converting Probots `LogggerWithTarget` to Bunyans `Logger`
- There is also new serializers to hash some(not yet all) sensitive data (UGC/PII)

**Whats next**
- There are still UGC/PII that aren't being hashed or removed, so will find them and update them appropriately.

### CODE REVIEW

**Review by gxueatlassian — APPROVED** [review]

File: src/config/logger.ts [hunk 1]
```diff
@@ -22,9 +22,10 @@ const LOG_STREAM = filteringHttpLogsStream(FILTERING_FRONTEND_HTTP_LOGS_MIDDLEWA
 	bformat({ outputMode, levelInString: true })
 );
 
+// Todo existing config serializer is throwing occasional error, investigate.
 const responseSerializer = (res: AxiosResponse) => ({
 	...stdSerializers.res(res),
-	config: JSON.parse(util.inspect(res?.config)), // removes circular dependency in json
+	config: JSON.parse(inspect(res?.config)), // removes circular dependency in json
```
Comment (mboudreau) [inline src/config/logger.ts:28]: wait, I'm fairly sure this is wrong.  I have some ideas on how to improve the logger here, might be simpler if we just go on a call I think.

File: src/config/logger.ts [hunk 2]
```diff
@@ -45,6 +46,22 @@ const errorSerializer = (err) => (!err || !err.stack) ? err : {
 	stack: getFullErrorStack(err)
 };
 
+const hashSerializer = (data: any): string => {
+	if (data === undefined || data == null) {
+		return data;
```
Comment (mboudreau) [inline src/config/logger.ts:51]: this should return `undefined` and not null so it gets removed from the payload when stringifying it.

File: src/config/logger.ts [hunk 3]
```diff
@@ -45,6 +46,22 @@ const errorSerializer = (err) => (!err || !err.stack) ? err : {
 	stack: getFullErrorStack(err)
 };
 
+const hashSerializer = (data: any): string => {
```
Comment (mboudreau) [inline src/config/logger.ts:49]: return type should be `string | undefined`

**Review by mboudreau — DISMISSED** [review]

**Review by gxueatlassian — DISMISSED** [review]

File: src/config/logger.ts [hunk 4]
```diff
@@ -45,6 +46,22 @@ const errorSerializer = (err) => (!err || !err.stack) ? err : {
...
+	if (data === undefined || data == null) {
+		return data;
+	}
+	return createHashWithSharedSecret(data);
```
[the remark this answers is not in the document]
Comment (joshkay10) [reply src/config/logger.ts:50]: Would be nice, then I can drop the hashserializer and call this directly!! However the `createHashWithSharedSecret` is expected to return a string, so It gets a bit messy trying to get around that 😒

**Review by gxueatlassian — APPROVED** [review]

**Pull request merged**